frontend/packages/dev-console/src/components/topology-list/TopologyListPage.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import Helmet from 'react-helmet';
+import { getActiveNamespace } from '@console/internal/actions/ui';
+import { TopologyIcon } from '@patternfly/react-icons';
+import { Firehose, StatusBox, FirehoseResult } from '@console/internal/components/utils';
+import { TransformResourceData, getResourceList } from '@console/shared';
+import { ProjectOverview } from '@console/internal/components/overview/project-overview';
+import { groupItems } from '@console/internal/components/overview';
+import { DeploymentKind } from '@console/internal/module/k8s';
+import NamespacedPage from '../NamespacedPage';
+import { EmptyMsg } from '../topology/TopologyPage';
+import TopologyToggleIcon from './TopologyToggleIcon';
+
+import './TopologyListPage.scss';
+
+interface TopologyListDataResource {
+  deployments: FirehoseResult<DeploymentKind[]>;
+  deploymentConfigs: FirehoseResult;
+  statefulSets: FirehoseResult;
+  daemonSets: FirehoseResult;
+}
+interface TopologyListControllerProps {
+  resources?: TopologyListDataResource;
+  loadError?: string;
+  loaded?: boolean;
+}
+
+const skeletonOverview = (
+  <div className="skeleton-overview">
+    <div className="skeleton-overview--head" />
+    <div className="skeleton-overview--tile" />
+    <div className="skeleton-overview--tile" />
+    <div className="skeleton-overview--tile" />
+  </div>
+);
+
+const TopologyListController: React.FC<TopologyListControllerProps> = (props) => {
+  const { resources, loadError, loaded } = props;
+
+  const transformResourceData = new TransformResourceData(props);
+  const items = [
+    ...transformResourceData.createDeploymentItems(resources.deployments.data),
+    ...transformResourceData.createDaemonSetItems(resources.daemonSets.data),
+    ...transformResourceData.createDeploymentConfigItems(resources.deploymentConfigs.data),
+    ...transformResourceData.createStatefulSetItems(resources.statefulSets.data),
+  ];
+
+  return (
+    <StatusBox
+      skeleton={skeletonOverview}
+      data={items}
+      label="Topology"
+      loaded={loaded}
+      loadError={loadError}
+      EmptyMsg={EmptyMsg}
+    >
+      <ProjectOverview groups={groupItems(items, '#GROUP_BY_APPLICATION#')} />
+    </StatusBox>
+  );
+};
+
+const TopologyListPageController: React.FC = () => {
```

Comment:
Should only have one component per-file in the `dev-console`. Files have low cost, and it allows easier maintenance.